tests/acceptance/stepDefinitions/filesContext.js

```diff
@@ -10,7 +10,76 @@ const util = require('util')
 let deletedElements
 let timeOfLastDeleteOperation = Date.now()
 let timeOfLastUploadOperation = Date.now()
-const { download } = require('../helpers/webdavHelper')
+const convert = require('xml-js')
+const { parts } = require('../helpers/path')
+const { download, propfind } = require('../helpers/webdavHelper')
+
+const getAllFilesFolders = function () {
+  const user = client.globals.currentUser
+  return propfind(`/files/${user}`, user, [])
+    .then(str => {
+      const data = convert.xml2js(str, { compact: true })['d:multistatus']['d:response']
+      const filenames = data.map(elements =>
+        parts(
+          elements['d:href']._text)
+      )
+      return filenames
+    })
+}
+
+const getAllFolders = async function () {
+  const filenames = await getAllFilesFolders()
+  const folderPaths = filenames.filter(elements => elements.includes(''))
+  const folders = []
+  folderPaths.forEach(arr => {
+    arr.pop()
+    folders.push(arr.pop())
+  })
+  // The first element in the folders is the username which is not actually a folder but relative path to every files and
+  //  folders.So the username which is always the first element needs to be removed from folders list
```

Comment:
Please write comments concisely. It's too long. You don't need to say, that you removed the first element. Just say, why that needs to be done. The `what` part is pretty apparent from the following line of code.